src/cephadm/cephadm.py

```diff
@@ -5422,12 +5422,20 @@ def prepare_ssh(
     if not ctx.skip_monitoring_stack:
         for t in ['ceph-exporter', 'prometheus', 'grafana', 'node-exporter', 'alertmanager']:
             logger.info('Deploying %s service with default placement...' % t)
-            cli(['orch', 'apply', t])
+            try:
+                cli(['orch', 'apply', t])
+            except Exception:
+                logger.error(f'Failed to apply service type {t}. '
+                             'Perhaps the ceph version being bootstrapped does not support it')
 
     if ctx.with_centralized_logging:
         for t in ['loki', 'promtail']:
             logger.info('Deploying %s service with default placement...' % t)
-            cli(['orch', 'apply', t])
+            try:
+                cli(['orch', 'apply', t])
+            except Exception:
+                logger.error(f'Failed to apply service type {t}. '
+                             'Perhaps the ceph version being bootstrapped does not support it')
```

Comment:
should we exit with an error user code in this case as one of the options provided by the user are failing? .. capturing the exception will lead to a "good" bootstrap when not all the provided options are not fulfilled.

Reply:
changed it to now set `ctx.error_code` when one of these failures happens, which is the same mechanism being used currently for returning a nonzero error code when applying a provided spec fails.